framework/doc/content/sqa/framework_scs.md

```diff
@@ -153,12 +256,164 @@ All non-system includes should use quotes with a single space between `include`
+
+- Algorithms over data
+
+  ```C++
+  norm(v);
+  det(A);
+  project(u, v);
+  ```
+
+  If it takes an object as input, produces a value, and does not modify the object, make it a free function. This aligns with the STL design: containers own storage; algorithms are free.
+
+- Extensibility across types
+
+  ```C++
+  template<class T>
+  auto norm(const T & x);
+  ```
+
+  Free functions allow generic programming, ADL-based customization, extension to foreign types. Member functions cannot be added retroactively.
+
+- Operations that conceptually sit “between” types
+
+  ```C++
+  intersect(mesh, plane);
+  assemble(system, element);
+  ```
+
+  These don’t belong to either object alone.
+
+In summary, when making a decision, ask these questions in order:
+
+1. Does this operation mutate the object or enforce invariants? Yes → member
+2. Is it conceptually “owned” by the object? Yes → member
+3. Is the operation symmetric or algorithmic? Yes → free function
+4. Do I want/expect others to extend this behavior for new types? Yes → free function
+5. If you’re unsure → default to free function.
+
+As a reminder, helper free functions defined only in the translation unit (i.e., `.C` files) should be marked as `static` or encapsulated in an anonymous namespace to force static linking.
+
+## Access control: public, protected, and private
+
+Choose the narrowest access level that correctly expresses intent.
+
+Public members and methods form the stable API of a class. They define what users are allowed to rely on. Use `public` only for:
+
+- Operations that are safe, well-defined, and intended for external use
+- Methods that preserve class invariants
+- Behavior you are willing to support long-term
+
+Public interfaces should be minimal, const-correct, and clearly documented. Adding something to `public` is a commitment.
+
+Prefer `private` by default. Use `private` for:
+
+- Representation details and internal state
+- Helper functions and implementation mechanics
+- Anything that should not be relied upon by derived classes or external code
+
+Private members allow internal refactoring without breaking user assumptions and are the strongest tool for enforcing invariants.
+
+`protected` exists to support controlled customization via inheritance, not general access. Use `protected` only when:
+
+- A derived class must override or extend behavior
+- The base class explicitly documents a customization or extension point
+- The invariant contract between base and derived classes is clear
+- Avoid exposing data members as `protected`. Prefer `protected` virtual functions or `protected` helper methods instead.
+
+If a member is not intended to be used or overridden by derived classes, it should **not** be `protected`.
+
+In summary:
+
+- Default to `private`
+- Promote to `public` only when there is a clear, justified use case
+- Treat `protected` as part of the class’s internal design contract, not its public API
```

Comment:
Most people, including me, don't know what "internal design contract" means

Reply:
Yeah, I rephrased it.